### PR TITLE
KMP - added python code(String_Search.mdx)

### PR DESCRIPTION
**Pull request by Nithya-02**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: content/6_Advanced/String_Search.mdx [hunk 1]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
 ```
 
 </CPPSection>
+<PySection>
+
+```py
+def solve(s):
```
Comment (envyaims) [inline content/6_Advanced/String_Search.mdx:105]:
```suggestion
def pi(s):
```

File: content/6_Advanced/String_Search.mdx [hunk 2]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
...
+	j = 0
+	for i in range(1,n):
+		while(j > 0 and s[j]!=s[i]):
+			j = pi[j-1]
+		if s[i] == s[j]:
+			j+=1
+		pi[i] = j
+	return pi
+	
+def KMP(s,t):
```
Comment (envyaims) [inline content/6_Advanced/String_Search.mdx:117]: whats this function about?

File: content/6_Advanced/String_Search.mdx [hunk 3]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
...
 
 </CPPSection>
+<PySection>
+
+```py
+def solve(s):
+	n = len(s)
+	pi = [0]*n
+	j = 0
+	for i in range(1,n):
```
Comment (envyaims) [inline content/6_Advanced/String_Search.mdx:109]: make spacing consistent and space out operators

**Review by envyaims — CHANGES_REQUESTED** [review]

File: content/6_Advanced/String_Search.mdx [hunk 4]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
...
+		pi[i] = j
+	return pi
+	
+def KMP(s,t):
+	p = t + "#" + s
+	ans = 0
+	for i in solve(p):
+		if i == len(t):
+			ans += 1
+	return ans
```
Comment (Aadit-Ambadkar) [inline content/6_Advanced/String_Search.mdx:123]:
```suggestion
```

File: content/6_Advanced/String_Search.mdx [hunk 5]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
 ```
 
 </CPPSection>
+<PySection>
+
+```py
+def solve(s):
+	n = len(s)
+	pi = [0]*n
```
Comment (Aadit-Ambadkar) [inline content/6_Advanced/String_Search.mdx:107]:
```suggestion
	pi = [0] * n
```

File: content/6_Advanced/String_Search.mdx (hunk 3 above)
Comment (Aadit-Ambadkar) [reply content/6_Advanced/String_Search.mdx:109]:
```suggestion
	for i in range(1, n):
```

File: content/6_Advanced/String_Search.mdx [hunk 6]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
...
+	j = 0
+	for i in range(1,n):
+		while(j > 0 and s[j]!=s[i]):
```
Comment (Aadit-Ambadkar) [inline content/6_Advanced/String_Search.mdx:110]:
```suggestion
		while(j > 0 and s[j] != s[i]):
```

File: content/6_Advanced/String_Search.mdx [hunk 7]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
...
+			j = pi[j-1]
```
Comment (Aadit-Ambadkar) [inline content/6_Advanced/String_Search.mdx:111]:
```suggestion
			j = pi[j - 1]
```

File: content/6_Advanced/String_Search.mdx [hunk 8]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
...
+		if s[i] == s[j]:
+			j+=1
```
Comment (Aadit-Ambadkar) [inline content/6_Advanced/String_Search.mdx:113]:
```suggestion
			j += 1
```

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: content/6_Advanced/String_Search.mdx [hunk 9]
```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
...
+		pi[i] = j
+	return pi
+	
+s = input()
+t = input()
+p = t + "#" + s
+pi = solve(p)
+ans = 0
```
Comment (brad182) [inline content/6_Advanced/String_Search.mdx]: could you add a few comments to make it more clear to readers

**Review by brad182 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/String_Search.mdx (hunk 2 above)
Comment (Nithya-02) [reply content/6_Advanced/String_Search.mdx:131]: This Function basically acts as a main function of the question. It takes 2 strings as an input and returns integer as a string.
(Written in the comments section)

**Review by Nithya-02 — COMMENTED** [review]
Reviewed All the changes

**Pull request closed without merging**